oxygen-account/src/main/java/com/oxygenxml/account/service/UserService.java

```diff
@@ -69,17 +102,75 @@ public User updateUser(User user) {
 		return userRepository.save(user);
 	}
 	
+	/**
+	 * Retrieves the currently authenticated user from the security context.
+	 * 
+	 * @return The authenticated User entity.
+	 */
+	public User getCurrentUser() {
+		
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		
+		if (authentication.getPrincipal() instanceof org.springframework.security.core.userdetails.User userPrincipal) {
+		        return userRepository.findByEmail(userPrincipal.getUsername());
+		}
+		
+		throw new UsernameNotFoundException(Message.INVALID_USER.getMessage());
```

Comment:
Exceptia ar trebui sa se numeasca `UserNotAuthenticatedException` iar mesajul ar trebui sa fie de forma:
`User not authenticated`